api/src/main/java/org/openmrs/util/DatabaseUtil.java

```diff
@@ -94,9 +93,80 @@ public static String loadDatabaseDriver(String connectionUrl, String connectionD
 	}
 	
 	/**
-	 * Executes the passed SQL query, enforcing select only if that parameter is set
+	 * Executes the passed SQL query, enforcing select only if that parameter is set for given Session
 	 */
-	public static List<List<Object>> executeSQL(Connection conn, String sql, boolean selectOnly) throws DAOException {
+	public static List<List<Object>> executeSQLWithSession(Session session, String sql, boolean selectOnly)
```

Comment:
The javadoc should state that "It is preferred to use session.createSQLQuery with correct synchronization (sqlQuery.addSynchronizedEntityClass) to invalidate session cache if necessary."